### ISSUE TITLE
mscash2-opencl done() releases unintialized resources

### ISSUE THREAD

**Issue by magnumripper** [magnumripper/JohnTheRipper#1609]
```
$ ../run/john --list=format-details -form:mscash*
mscash	81	512	98304	0102000f	11	MD4 32/64	MS Cache Hash (DCC)		16	44		0	root:176a4c2bd45ac73687676c2f09045353
mscash2	125	32	256	0102000f	15	PBKDF2-SHA1 128/128 AVX 4x	MS Cache Hash 2 (DCC2)		-1	16	260		0	bin:c0cbe0313a861062e29f92ede58f9b36
mscash-opencl	27	1	1	00020007	11	MD4 OpenCL	M$ Cache Hash		16	48		0	M$test2#ab60bdb4493822b175486810ac2abe63
mscash2-opencl	125	1	1	0002000f	17	PBKDF2-SHA1 OpenCL	MS Cache Hash 2 (DCC2)		-1	4	140		0	$DCC2$test#a86012faf7d88d1fc037a69764a92cac
ASAN:SIGSEGV
=================================================================
==79983==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7fff8d083a8d sp 0x7fff5dceb770 bp 0x7fff5dceb800 T0)
    #0 0x7fff8d083a8c in __dynamic_cast (in libc++abi.dylib) + 29
    #1 0x7fff8dc7a556 in clReleaseMemObject (in OpenCL) + 36
    #2 0x10227081f in releaseDevObjGws opencl_mscash2_helper_plug.c:92
    #3 0x102273922 in releaseAll opencl_mscash2_helper_plug.c:121
    #4 0x102270651 in done opencl_mscash2_fmt_plug.c:183
    #5 0x1023ed0a0 in fmt_done formats.c:103
    #6 0x102447499 in listconf_parse_late listconf.c:569
    #7 0x1023fca06 in john_init john.c:1318
    #8 0x1023fe8a2 in main john.c:1728
    #9 0x7fff8a7aa5c8 in start (in libdyld.dylib) + 0
    #10 0x2

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV ??:0 ??
==79983==ABORTING
```

**Comment by magnumripper** [comment]
Without ASan
```
$ ../run/john --list=format-details -form:mscash*
mscash	81	512	98304	0102000f	11	MD4 32/64	MS Cache Hash (DCC)		16	44		0	root:176a4c2bd45ac73687676c2f09045353
mscash2	125	32	256	0102000f	15	PBKDF2-SHA1 128/128 AVX 4x	MS Cache Hash 2 (DCC2)		-1	16	260		0	bin:c0cbe0313a861062e29f92ede58f9b36
mscash-opencl	27	1	1	00020007	11	MD4 OpenCL	M$ Cache Hash		16	48		0	M$test2#ab60bdb4493822b175486810ac2abe63
mscash2-opencl	125	1	1	0002000f	17	PBKDF2-SHA1 OpenCL	MS Cache Hash 2 (DCC2)		-1	4	140		0	$DCC2$test#a86012faf7d88d1fc037a69764a92cac
OpenCL error (CL_INVALID_KERNEL) in file (opencl_mscash2_helper_plug.c) at line (123) - (Error releasing kernel pbkdf2_preprocess_short)
```

**Comment by magnumripper** [comment]
Fixed in d647405ee
